autoload/go/cmd.vim

```diff
@@ -117,16 +117,16 @@ endfunction
 
 " Run runs the current file (and their dependencies if any) in a new terminal.
 function! go#cmd#RunTerm(bang, mode, files) abort
-  let cmd = "go run "
+  let cmd = ["go", "run"]
   let tags = go#config#BuildTags()
-  if len(tags) > 0
-    let cmd .= "-tags " . go#util#Shellescape(tags) . " "
+  if go#config#BuildTags() > 0
+    call extend(cmd, ["-tags", go#util#Shellescape(tags)])
```

Comment:
Remove the `go#util#Shellescape` call; `tags` alone should be fine since `cmd` won't be joined into a string anymore.